### PR TITLE
#7420 - compress images UWP fix

### PR DESCRIPTION
**Pull request by eszterhorvath**
Fixes #7420.

Changes Proposed in this pull request:
- Images are now compressed on UWP too.
- If "Save pictures to gallery" is enabled and the inspection's name is not a valid folder name, the invalid characters are removed.

### CODE REVIEW

File: src/Media.Plugin/UWP/NewMediaImplementation.cs [hunk 1]
```diff
@@ -440,8 +443,6 @@ Task<bool> ResizeAsync(StorageFile file, StoreCameraMediaOptions mediaOptions)
                 if (photoSize == PhotoSize.Full)
                     return Task.FromResult(false);
 
-				var customPhotoSize = mediaOptions.CustomPhotoSize;
-				var quality = mediaOptions.CompressionQuality;
```
Comment (eszterhorvath) [inline src/Media.Plugin/UWP/NewMediaImplementation.cs:444]: It was never used.

Comment (eszterhorvath) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:444]: And customPhotoSize was only used once too, in line 462.

Comment (bruzkovsky) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:444]: the quality?

Comment (eszterhorvath) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:444]: The quality variable, mediaOptions.CompressionQuality is used instead.

File: src/Media.Plugin/UWP/NewMediaImplementation.cs [hunk 2]
```diff
@@ -200,8 +200,11 @@ async Task SaveToAlbum(StorageFile file, StorageFolder folder, string directory
         {
             try
             {
-                if (IsValidPathName(directory))
-                    folder = await folder.CreateFolderAsync(directory, CreationCollisionOption.GenerateUniqueName);
+                if (!IsValidFileName(directory))
+                {
+                    directory = string.Join("", directory.Split(Path.GetInvalidFileNameChars()));
+                }
+                folder = await folder.CreateFolderAsync(directory, CreationCollisionOption.GenerateUniqueName);
```
Comment (bruzkovsky) [inline src/Media.Plugin/UWP/NewMediaImplementation.cs:207]: `directory` could be `null` or `string.Empty` at this point

File: src/Media.Plugin/UWP/NewMediaImplementation.cs [hunk 3]
```diff
@@ -200,8 +200,11 @@ async Task SaveToAlbum(StorageFile file, StorageFolder folder, string directory
         {
             try
             {
-                if (IsValidPathName(directory))
-                    folder = await folder.CreateFolderAsync(directory, CreationCollisionOption.GenerateUniqueName);
+                if (!IsValidFileName(directory))
+                {
+                    directory = string.Join("", directory.Split(Path.GetInvalidFileNameChars()));
```
Comment (bruzkovsky) [inline src/Media.Plugin/UWP/NewMediaImplementation.cs:205]: `directory` could be `null` or `string.Empty` at this point

**Review by bruzkovsky — CHANGES_REQUESTED** [review]

File: src/Media.Plugin/UWP/NewMediaImplementation.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (eszterhorvath) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:205]: The `directory` contains only the name of the new directory, not a path (the `folder` stores that). In line 207 `folder.CreateFolderAsync` gets `directory` as its first parameter which is "desiredName".
My original problem was that the `directory` variable contained slashes, that caused the problem and a path can contain slashes so here we need to check the invalid file name chars and not the path chars.

File: src/Media.Plugin/UWP/NewMediaImplementation.cs (hunk 2 above)
Comment (eszterhorvath) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:207]: Oh true. And if so, should I just put the image to the Saved Pictures folder without a subfolder?

Comment (bruzkovsky) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:207]: exactly

File: src/Media.Plugin/UWP/NewMediaImplementation.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (bruzkovsky) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:205]: ah, ok I get it now! Then it's fine

Comment (bruzkovsky) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:204]: you missed this one ;-)

Comment (eszterhorvath) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:204]: oooh

**Review by bruzkovsky — APPROVED** [review]

File: src/Media.Plugin/UWP/NewMediaImplementation.cs [hunk 4]
```diff
@@ -481,6 +483,10 @@ Task<bool> ResizeAsync(StorageFile file, StoreCameraMediaOptions mediaOptions)
                         var finalWidth = Convert.ToUInt32(bitmap.PixelWidth * percent);
                         var finalHeight = Convert.ToUInt32(bitmap.PixelHeight * percent);
 
+                        var folder = await file.GetParentAsync();
```
Comment (gentledepp) [inline src/Media.Plugin/UWP/NewMediaImplementation.cs:486]: this alone solves the issue?
Deleting the file and creating a new one? :-)

Comment (bruzkovsky) [reply src/Media.Plugin/UWP/NewMediaImplementation.cs:486]: jep... 🙈

**Pull request merged**